repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -3585,6 +3478,26 @@ public List<String> propagateClassification(String entityGuid, String classifica
             AtlasVertex currentTaskVertex = (AtlasVertex) graph.query().has(TASK_GUID, currentTask.getGuid()).vertices().iterator().next();
             currentTaskVertex.setProperty(TASK_ASSET_COUNT_TO_PROPAGATE, impactedVertices.size());
 
+
+            //iterate over each impacted vertex to create a message and send it to kafka
+            for (AtlasVertex vertex : impactedVertices) {
+                // Create a Map to store vertex properties
+                Map<String, Object> vertexMap = new HashMap<>();
+                ObjectMapper objectMapper = new ObjectMapper(); // Jackson ObjectMapper
+
+                vertexMap.put("parentTaskVertexId", vertex.getProperty("parentTaskVertexId", String.class));
+                vertexMap.put("action", "ADD_PROPAGATION");
+                vertexMap.put("assetVertexId", vertex.getIdForDisplay());
+                vertexMap.put("tagVertexId", classificationVertexId);
+                vertexMap.put("parentTaskGuid", currentTask.getGuid());
+                vertexMap.put("assetGuid", vertex.getIdForDisplay());
+                vertexMap.put("tagTypeName", currentTask.getClassificationTypeName());
+
+                String vertexJson = objectMapper.writeValueAsString(vertexMap);
+
+                //send vertexJson to kafka topic 'TAG_PROP_EVENTS'
+                kfknotif.sendInternal(NotificationInterface.NotificationType.RELATIONSHIPS, Collections.singletonList(vertexJson));
```

Comment:
Do we not have to mention the topic name for this method to work? It looks like a pretty generic call.

Reply:
I misunderstood the notification type, I dug in a bit deeper and found out that the notification type is responsible to find the correct producer/consumer and use it. Added my enum there.